packages/loot-core/src/server/accounts/rules.ts

```diff
@@ -811,28 +811,28 @@ export class Rule {
 }
 
 export class RuleIndexer {
-  field;
-  method;
-  rules;
+  field: string;
+  method?: string;
+  rules: Map<string, Set<Rule>>;
 
   constructor({ field, method }: { field: string; method?: string }) {
     this.field = field;
     this.method = method;
     this.rules = new Map();
   }
 
-  getIndex(key) {
+  getIndex(key: string): Set<Rule> {
     if (!this.rules.has(key)) {
       this.rules.set(key, new Set());
     }
     return this.rules.get(key);
   }
 
-  getIndexForValue(value) {
+  getIndexForValue(value: unknown): Set<Rule> {
     return this.getIndex(this.getKey(value) || '*');
   }
 
-  getKey(value) {
+  getKey(value: unknown): string | null {
```

Comment:
Looking at `rule.d.ts`, it seems like the value is always a `string`, `number`, `string[]`, or `number[]`. What do you think about either explicitly enumerating those types here, or using `BaseConditionEntity<keyof FieldValueTypes, RuleConditionOp>['value']`? The latter one would be nice since it would support any new types we add in the future.

Reply:
The function doesn't realy care what it is passed. So I think leaving it broarder work better?

Reply:
Sure, it probably doesn't matter much in this case